### PR TITLE
Add tests for completion reader in JdbcLedgerDaoSpec

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala [hunk 1]
```diff
@@ -21,6 +22,28 @@ private[dao] trait JdbcLedgerDaoCompletionsSpec extends OptionValues {
...
+    for {
+      from <- ledgerDao.lookupLedgerEnd()
+      (offset, tx) <- storeCreateTransaction()
+      to <- ledgerDao.lookupLedgerEnd()
+      (_, response) <- ledgerDao.completions
+        .getCommandCompletions(from, to, tx.applicationId.get, Set(tx.submittingParty.get))
+        .runWith(Sink.head)
+    } yield {
+      val receivedOffset =
+        ApiOffset.assertFromString(response.checkpoint.value.offset.value.value.absolute.value)
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: Please pull this out into a helper, with a meaningful name.

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  private def rejectWith(reason: RejectionReason): PersistenceEntry.Rejection =
+    PersistenceEntry.Rejection(
+      LedgerEntry.Rejection(
+        recordTime = Instant.now,
+        commandId = UUID.randomUUID().toString,
+        applicationId = applicationId,
+        submitter = party,
+        rejectionReason = reason,
+      )
+    )
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: Can we move this stuff to the bottom of this file, or a companion object if possible?

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      RejectionReason.OutOfQuota(""),
+      RejectionReason.PartyNotKnownOnLedger(""),
+      RejectionReason.SubmitterCannotActViaParticipant(""),
+      RejectionReason.TimedOut(""),
+    )
+
+    for {
+      from <- ledgerDao.lookupLedgerEnd()
+      _ <- Future.sequence(
+        for (reason <- reasons; offset = nextOffset())
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: You can probably inline `offset`. Even if you don't, I think it'd be clearer if it lived in the `yield` block.

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      (_, response) <- ledgerDao.completions
+        .getCommandCompletions(from, to, applicationId, parties)
+        .runWith(Sink.head)
+    } yield {
+      val receivedOffset =
+        ApiOffset.assertFromString(response.checkpoint.value.offset.value.value.absolute.value)
+      receivedOffset shouldBe offset
+
+      response.completions should have length 1
+      val completion = response.completions.head
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]:
```suggestion
      val completion = response.completions.loneElement
```

Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: For future reference, this is added by the trait `LoneElement`.

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/5072/commits/2f35ea00e91ac6dd081dc601dfd4ece98c9923b7

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/5072/commits/5a84f21e6031c81f89c9a7ce87e099ac3e82a72b

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala (hunk 3 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/5072/commits/a18bf358d4c7d743d9738aee6fb84c70da1d0afd

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala (hunk 4 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoCompletionsSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/5072/commits/3785e7c393670e0a218c4680b173a8d3ad34cbf2

**Pull request merged**